screens/PokemonList.tsx

```diff
@@ -1,29 +1,100 @@
 import React, { useEffect, useState } from 'react';
-import { StyleSheet, Text, View, Button } from 'react-native';
-
-const PokemonList = ({ navigation }) => {
-  return (
-    <View style={styles.container}>
-      <Text style={styles.header}>Pokemon List</Text>
-      {/* <Button
-        title='Go to Native Module'
-        onPress={() => navigation.navigate('NativeModule')}
-      /> */}
-    </View>
-  );
+import { StyleSheet, Text, View, FlatList, Dimensions } from 'react-native';
+import { getAllPokemons } from '../api/pokemon';
+
+const keyExtractor = ({ name }) => name;
+
+const PokemonList = () => {
+  const step = 20;
+  const [batchStart, setBatchStart] = useState(0);
+  const [loading, setLoading] = useState(true);
+  const [pokemons, setPokemons] = useState(null);
+
+  useEffect(() => {
+    const effect = async () => {
+      const allPokemons = await getAllPokemons(batchStart, step);
+      setPokemons(allPokemons);
+      setLoading(false);
+      setBatchStart(batchStart + step);
+    };
+    effect();
+  }, []);
+
+  const renderItem = ({ item }) => {
+    //const matchRes = item.url.match('(pokemon\\/{1}(\\d+)\\/{1})', 'gi');
+
+    // this is more readable than regex
+    const urlChopped = item.url.split('/');
+    const id = urlChopped[urlChopped.length - 2];
+    return (
+      <Text style={styles.header}>
+        {id}. {item.name}
+      </Text>
+    );
+  };
+
+  const scrolledCloseToEnd = ({
+    layoutMeasurement,
+    contentOffset,
+    contentSize
+  }) => {
+    return (
+      layoutMeasurement.height + contentOffset.y >= contentSize.height - 200
+    );
+  };
+
+  const getMorePokemons = async () => {
+    const morePokemons = await getAllPokemons(batchStart, step);
+    setPokemons([...pokemons, ...morePokemons]);
+    setBatchStart(batchStart + step);
+  };
+
+  if (loading) {
+    return (
+      <View style={styles.container}>
+        <Text style={styles.pokemonListElement}>Loading...</Text>
+      </View>
+    );
+  } else {
+    return (
+      <View style={styles.container}>
+        <Text style={styles.header}>Pokemon List</Text>
+
+        <FlatList
+          style={styles.list}
+          data={pokemons}
+          renderItem={renderItem}
+          keyExtractor={keyExtractor}
+          onScroll={({ nativeEvent }) => {
```

Comment:
I think there's `onEndReached` which you could use instead (but it's good that you played with `onScroll`! 💯)

Reply:
Apparently there's a problem with onEndReached. It gets called on render :/

Reply:
Could you elaborate what the problem is (here or on Slack)? It's quite "normal" to use it for refetching :) there's also `onEndReachedThreshold` for better control when exactly it's called